examples/linear_model/plot_tweedie_regression_insurance_claims.py

```diff
@@ -0,0 +1,594 @@
+# Here we model the frequency ``y = ClaimNb / Exposure``, which is still a
+# (scaled) Poisson distribution, and use ``Exposure`` as `sample_weight`.
+
+df_train, df_test, X_train, X_test = train_test_split(df, X, random_state=0)
+
+# The parameters of the model are estimated by minimizing the Poisson deviance
+# on the training set via a quasi-Newton solver: l-BFGS. Some of the features
+# are collinear, we use a weak penalization to avoid numerical issues.
+glm_freq = PoissonRegressor(alpha=1e-3, max_iter=400)
+glm_freq.fit(X_train, df_train["Frequency"],
+             sample_weight=df_train["Exposure"])
+
+scores = score_estimator(
+    glm_freq,
+    X_train,
+    X_test,
+    df_train,
+    df_test,
+    target="Frequency",
+    weights="Exposure",
+)
+print(scores)
+
+##############################################################################
+#
+# We can visually compare observed and predicted values, aggregated by the
+# drivers age (``DrivAge``), vehicle age (``VehAge``) and the insurance
+# bonus/malus (``BonusMalus``).
+
+fig, ax = plt.subplots(ncols=2, nrows=2, figsize=(16, 8))
+fig.subplots_adjust(hspace=0.3, wspace=0.2)
+
+plot_obs_pred(
+    df=df_train,
+    feature="DrivAge",
+    weight="Exposure",
+    observed="Frequency",
+    predicted=glm_freq.predict(X_train),
+    y_label="Claim Frequency",
+    title="train data",
+    ax=ax[0, 0],
+)
+
+plot_obs_pred(
+    df=df_test,
+    feature="DrivAge",
+    weight="Exposure",
+    observed="Frequency",
+    predicted=glm_freq.predict(X_test),
+    y_label="Claim Frequency",
+    title="test data",
+    ax=ax[0, 1],
+    fill_legend=True
+)
+
+plot_obs_pred(
+    df=df_test,
+    feature="VehAge",
+    weight="Exposure",
+    observed="Frequency",
+    predicted=glm_freq.predict(X_test),
+    y_label="Claim Frequency",
+    title="test data",
+    ax=ax[1, 0],
+    fill_legend=True
+)
+
+plot_obs_pred(
+    df=df_test,
+    feature="BonusMalus",
+    weight="Exposure",
+    observed="Frequency",
+    predicted=glm_freq.predict(X_test),
+    y_label="Claim Frequency",
+    title="test data",
+    ax=ax[1, 1],
+    fill_legend=True
+)
+
+
+##############################################################################
+#
+# According to the observed data, the frequency of accidents is higher for
+# drivers younger than 30 years old, and it positively correlated with the
+# `BonusMalus` variable. Our model is able to mostly correctly model this
+# behaviour.
+#
+# Severity Model -  Gamma distribution
+# ------------------------------------
+# The mean claim amount or severity (`AvgClaimAmount`) can be empirically
+# shown to follow approximately a Gamma distribution. We fit a GLM model for
+# the severity with the same features as the frequency model.
+#
+# Note:
+#
+# - We filter out ``ClaimAmount == 0`` as the Gamma distribution has support
+#   on :math:`(0, \infty)`, not :math:`[0, \infty)`.
+# - We use ``ClaimNb`` as `sample_weight`.
```

Comment:
Why?

Reply:
In short: Every claim is an observation.
If you have 2 claims, say 1000 and 2000, then your average claim amount, alias severity, is 1500. If another person has 1 claim, say 3000, the correct average over all claims is (1000+2000+3000)/3 = (2 * 1500 + 1 * 3000)/(2+1) = 2000. If you had not used the claim number as sample weight, you would get the wrong average, i.e. (1500+3000)/2=2250.

Reply:
Right, that makes total sense once you understand the example, I was still a bit confused in my mind. Thanks!

Reply:
Side question, would it be equivalent to multiply the column `ClaimAmount` by `ClaimNb`?

Reply:
?What do you mean? Side answer: `ClaimAmount = ClaimNb * AvgClaimAmount` for each row/policy.